core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/retry/SdkDefaultRetryStrategy.java

```diff
@@ -165,6 +167,10 @@ public static AdaptiveRetryStrategy.Builder adaptiveRetryStrategyBuilder() {
         if (maxAttempts != null) {
             builder.maxAttempts(maxAttempts);
         }
+        if (builder instanceof BaseRetryStrategy.Builder) {
+            BaseRetryStrategy.Builder b = (BaseRetryStrategy.Builder) builder;
+            b.markDefaultAdded(DEFAULTS_NAME);
+        }
```

Comment:
BaseRetryStrategy is an internal API, so we shouldn't use it from here or anywhere else outside of the module. Should we push up the BaseRetryStrategy methods we're using here, or migrate the base to be a protected API? I feel like the former, but I'm easy to convince.

Reply:
Right now `BaseRetryStrategy.Builder` has no parent, so, yeah, we could create a `DefaultRetryStrategyBuilder` interface that is protected API for the methods used here, and cast to that type instead of `BaseRetryStrategy.Builder`, I think this would be clean and would work with the current hierarchy of the Builder inheritance which is tricky to work with already.

Reply:
wait the `BaseRetryStrategy.Builder` is already marked `@SdkProtectedApi`

Reply:
Decided to name it `DefaultAwareRetryStrategy` since we already have a [DefaultRetryStrategy](https://github.com/aws/aws-sdk-java-v2/blob/master/core/retries/src/main/java/software/amazon/awssdk/retries/DefaultRetryStrategy.java) class